.github/actions/run-upgrade-e2e/action.yaml

```diff
@@ -0,0 +1,188 @@
+name: "Run Upgrade E2E"
```

Comment:
Why does it need a separate action so similar to the existing run-e2e? Couldn't it just use the existing one with some conditionals? Or share some common parts?

Reply:
Ok, makes sense. Can the common actions be extracted though? "Determine base ref and image version", "Tweak node and Kubernetes", "Deploy base scylla-operator" etc look identical.

Reply:
Extracted common steps